pages/_app.js

```diff
@@ -35,16 +35,24 @@ function DotCMSStatus({ status }) {
     );
 }
 
-function RoutedComponent({ Component, pageRender, nav, isBeingEditFromDotCMS, language, location: { pathname }, pageProps }) {
+function RoutedComponent({
+    Component,
+    pageRender,
+    nav,
+    isBeingEditFromDotCMS,
+    language,
+    location: { pathname },
+    pageProps
+}) {
     const isFirstRun = useRef(true);
     const [requestedPage, setRequestedPage] = useState(null);
     const [clientRequestError, setClientRequestError] = useState(null);
     const [lang, setLang] = useState(language.current);
 
-    language.set  = (event) => {
-        language.current = event.target.value
-        setCookie('dotSPALang', event.target.value);
-        setLang(event.target.value);
+    language.set = (value) => {
+        language.current = value;
```

Comment:
is this needed? (maybe I'm wrong)
I know on line 50 you get that from the server, but on this `set`, it only gets called from the UI...

Reply:
Yes, because then it get passed in the `Provider` and then use it in the consumer here: https://github.com/dotCMS/dotcms-spa/pull/67/files#diff-a7ee9cdeb3cb81871dd684f9a4216fbcR36

I don't like that approach that much, but I wasn't able to come up with something better.

Ideas?